ci/downstream.sh

```diff
@@ -0,0 +1,221 @@
+#!/bin/bash -eu
+
+# Script to dual-home the upstream and downstream Collection in a single repo
+#
+#   This script will build or test a downstream collection, removing any
+#   upstream components that will not ship in the downstream release
+#
+#   NOTES:
+#       - All functions are prefixed with f_ so it's obvious where they come
+#         from when in use throughout the script
+
+DOWNSTREAM_VERSION="1.0.0"
+KEEP_DOWNSTREAM_TMPDIR="${KEEP_DOWNSTREAM_TMPDIR:-''}"
+INSTALL_DOWNSTREAM_COLLECTION_PATH="${INSTALL_DOWNSTREAM_COLLECTION_PATH:-}"
+_build_dir=""
+
+# Get the absolute path of the script directory
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOT_DIR="${SCRIPT_DIR}/.."
+
+f_log_info()
+{
+    printf "%s:LOG:INFO: %s\n" "${0}" "${1}"
+}
+
+f_show_help()
+{
```

Comment:
style: while this is valid a lot of the bash we write uses the syntax below. go with whatever the overall style approach so this is more of an observation.

```suggestion
f_show_help() {
```